src/WindowsUtils/BusyLoopUtils.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "WindowsUtils/BusyLoopUtils.h"
+
+#include <absl/base/casts.h>
+#include <absl/strings/str_format.h>
+
+#include "OrbitBase/GetLastError.h"
+#include "WindowsUtils/ReadProcessMemory.h"
+#include "WindowsUtils/SafeHandle.h"
+#include "WindowsUtils/WriteProcessMemory.h"
+
+namespace orbit_windows_utils {
+
+namespace {
+
+ErrorMessage GetLastErrorForFunction(std::string_view function) {
+  return ErrorMessage(
+      absl::StrFormat("Calling \"%s\": %s", function, orbit_base::GetLastErrorAsString()));
+}
+
+ErrorMessageOr<void> FlushInstructionCache(HANDLE process_handle, void* address, size_t size) {
+  if (::FlushInstructionCache(process_handle, address, size) == 0) {
+    return GetLastErrorForFunction("FlushInstructionCache");
+  }
+  return outcome::success();
+}
+
+ErrorMessageOr<void> WriteInstructionsAtAddress(const std::string& byte_buffer, void* address) {
+  // Mark memory page for writing.
+  DWORD old_protect = 0;
+  if (!VirtualProtect(address, byte_buffer.size(), PAGE_EXECUTE_READWRITE, &old_protect)) {
+    return GetLastErrorForFunction("VirtualProtect");
+  }
+
+  // Write original bytes.
+  memcpy(address, byte_buffer.data(), byte_buffer.size());
+
+  // Restore original page protection.
+  if (!VirtualProtect(address, byte_buffer.size(), old_protect, &old_protect)) {
+    return GetLastErrorForFunction("VirtualProtect");
+  }
+
+  OUTCOME_TRY(FlushInstructionCache(GetCurrentProcess(), address, byte_buffer.size()));
+  return outcome::success();
+}
+
+}  // namespace
+
+ErrorMessageOr<BusyLoopInfo> InstallBusyLoopAtAddress(HANDLE process_handle, void* address) {
+  constexpr const char kBusyLoopCode[] = {0xEB, 0xFE};
```

Comment:
This at least deserves a comment stating the instruction used.

Reply:
Added a comment.